src/tiled/stampbrush.cpp

```diff
@@ -386,6 +401,52 @@ void StampBrush::drawPreviewLayer(const QVector<QPoint> &list)
             mapDocument()->unifyTilesets(variation.map, mMissingTilesets);
 
             TileLayer *stamp = variation.tileLayer();
+            bool copyFlag = false;
+
+            //if staggered map, makes sure stamp stays the same
+            if ((mapDocument()->map()->orientation() == Map::Hexagonal
+                    || mapDocument()->map()->orientation() == Map::Staggered)
+                    && ((mapDocument()->map()->staggerAxis() == Map::StaggerY)?
+                         stamp->height() > 1 : stamp->width() > 1)) {
+                stamp = variation.tileLayer()->copy(variation.tileLayer()->bounds());
+                copyFlag = true;
```

Comment:
I don't think that is possible in an easy way. So at that point, the optimization because too much of an engineering effort and it's not worth doing it unless we notice an actual performance issue with the easy approach.